src/coreclr/jit/ssabuilder.cpp

```diff
@@ -248,40 +246,28 @@ void SsaBuilder::ComputeImmediateDom(BasicBlock** postOrder, int count)
 
             DBG_SSA_JITDUMP("Visiting in reverse post order: " FMT_BB ".\n", block->bbNum);
 
-            // Find the first processed predecessor block.
-            BasicBlock* predBlock = nullptr;
-            for (FlowEdge* pred = m_pCompiler->BlockPredsWithEH(block); pred; pred = pred->getNextPredEdge())
+            // Intersect DOM, if computed, for all predecessors.
+            BasicBlock* bbIDom = nullptr;
+            for (FlowEdge* pred = m_pCompiler->BlockDominancePreds(block); pred; pred = pred->getNextPredEdge())
             {
-                if (BitVecOps::IsMember(&m_visitedTraits, m_visited, pred->getSourceBlock()->bbNum))
+                BasicBlock* domPred = pred->getSourceBlock();
+                if ((domPred->bbPostorderNum >= (unsigned)count) || (postOrder[domPred->bbPostorderNum] != domPred))
                 {
-                    predBlock = pred->getSourceBlock();
-                    break;
+                    continue; // Unreachable pred
                 }
-            }
 
-            // There could just be a single basic block, so just check if there were any preds.
-            if (predBlock != nullptr)
-            {
-                DBG_SSA_JITDUMP("Pred block is " FMT_BB ".\n", predBlock->bbNum);
-            }
+                if ((numIters <= 0) && (domPred->bbPostorderNum <= (unsigned)i))
```

Comment:
This `<= 0` confused me for a second, maybe `== 0` ?

Reply:
You're not the first to be confused by my use of `<= 0` for these kinds of comparisons, so maybe I should switch to `==` in the future. But let me avoid rerunning CI for it here.